### DISCUSSION TITLE
Trades - export csv - identify taker and maker

### DISCUSSION THREAD

**Discussion by onlymarcus** [bisq-network/bisq#5799] [Ideas]
I have a suggestion for BISQ. When we export the CSV file with the trades carried out, the file does not have a column informing whether the taker was the buyer or the seller. This would help a lot for us to draw up buy and sell statistics
![imagem](https://user-images.githubusercontent.com/36059034/139937725-98589bdc-ee79-44c0-9c6e-aa121c1981f0.png)

**Comment by GordianLN** [comment]
I just expanded on this with #5804

**Reply by onlymarcus** [reply]
The percentage deviation from the current price is already in the csv file, it needs to include the identification of the trade maker and taker.

**Comment by GordianLN** [comment]
No it's not... not for me at least and I'm at the latest version, just checked

**Comment by pazza83** [comment]
Hi @chimp1984 this has been requested a few times. I know it was also recently discussed on a call also. Do you think the trades could be updated to show deviation from market price and also if the offer was Buy BTC or Sell BTC?

**Comment by chimp1984** [comment]
Yes, I can have a look to it. But we should be conservative with adding data to trade statistics as they are a lot (> 100 000) and come with some memory/performance panenties... The buy/sell is just a boolean so thats cheap to add. The price deviation is 8 bytes, so thats a about 10-15% increase of current data.
The other aspect is privacy. The more specific data we add the easier is correlation of offers to trade statistics (enables to derive that an offer was actually taken as trade). The percentage price is a quite specific value which adds up with the other values like price and amount to a more accurate matching chance.

**Comment by GordianLN** [comment]
I understand the reasoning about privacy; regarding the last sentence, it can also be derived by just monitoring available offers, and if an offer stops being available, and a matching offer appears in the history ;) Also if offer was maker or taker can be inferred by price of trade vs current price at moment of trade.
Regarding data, yes, it is indeed massive already as it is. I might add that a dialogue to choose "maximum N months backlog since..." could solve the problem; that dialogue could be made optional for "simple" export, and mandatory to up to 12? 24? months back if you also want deviation+maker/taker statistics.
At some point in time you will need to add that maximum export period anyway, given the increasing number of completed trades.

**Comment by pazza83** [comment]
Discussed with @chimp1984 adding both buy/sell as well as deviation from market price data to the trade statistics.

It was agreed this would not affect privacy too much.

Asked @jmacxx to take a look at this in a couple of weeks when he has the capacity to do so.

**Comment by pazza83** [comment]
Hi @jmacxx when looking into this please can you also consider if https://github.com/bisq-network/bisq/discussions/5830 can be included.
